### PR TITLE
2HDMa signal production for Run2 UL

### PR DESCRIPTION
**Pull request by shdutta16**
Changes have been made in the scripts to generate gridpacks for 2HDM+a signal MC for Run2 UL. This is for the MonoHiggs to gamma gamma analysis (SUS PAG). The datacards and the pythia fragmentation has been updated according to the  requirements of the analysis.

### CODE REVIEW

File: genfragments/ThirteenTeV/monoHiggs_UL/pythia8_hadronizer_nomatching_Haa_cff_CP5_PSweights_2018.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import FWCore.ParameterSet.Config as cms
+from Configuration.Generator.Pythia8CommonSettings_cfi import *
+from Configuration.Generator.MCTunes2017.PythiaCP5Settings_cfi import *
+from Configuration.Generator.PSweightsPythia.PythiaPSweightsSettings_cfi import *
+
+generator = cms.EDFilter("Pythia8HadronizerFilter",
```
Comment (menglu21) [inline genfragments/ThirteenTeV/monoHiggs_UL/pythia8_hadronizer_nomatching_Haa_cff_CP5_PSweights_2018.py:6]: it should be the concurrent one from UL

File: bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/monoHiggs/2HDMa_UL/2HDMa_bb_sinp_theta_tanb_beta_mXd_10_MH3_YY_MH4_XX_MH2_YY_MHC_YY/2HDMa_bb_sinp_theta_tanb_beta_mXd_10_MH3_YY_MH4_XX_MH2_YY_MHC_YY_run_card.dat [hunk 2]
```diff
@@ -0,0 +1,286 @@
...
+#*********************************************************************
+# Tag name for the run (one word)                                    *
+#*********************************************************************
+  tag_1	= run_tag ! name of the run 
+#*********************************************************************
+# Number of events and rnd seed                                      *
+# Warning: Do not generate more than 1M events in a single run       *
+# If you want to run Pythia, avoid more than 50k events in a run.    *
+#*********************************************************************
+  500000	= nevents ! Number of unweighted events requested
```
Comment (menglu21) [inline bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/monoHiggs/2HDMa_UL/2HDMa_bb_sinp_theta_tanb_beta_mXd_10_MH3_YY_MH4_XX_MH2_YY_MHC_YY/2HDMa_bb_sinp_theta_tanb_beta_mXd_10_MH3_YY_MH4_XX_MH2_YY_MHC_YY_run_card.dat:31]: change 500000 to a small number, e.g. 5000

File: bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/monoHiggs/2HDMa_UL/2HDMa_gg_sinp_theta_tanb_beta_mXd_10_MH3_YY_MH4_XX_MH2_YY_MHC_YY/2HDMa_gg_sinp_theta_tanb_beta_mXd_10_MH3_YY_MH4_XX_MH2_YY_MHC_YY_run_card.dat [hunk 3]
```diff
@@ -0,0 +1,286 @@
...
+#*********************************************************************
+# Tag name for the run (one word)                                    *
+#*********************************************************************
+  tag_1	= run_tag ! name of the run 
+#*********************************************************************
+# Number of events and rnd seed                                      *
+# Warning: Do not generate more than 1M events in a single run       *
+# If you want to run Pythia, avoid more than 50k events in a run.    *
+#*********************************************************************
+  500000	= nevents ! Number of unweighted events requested
```
Comment (menglu21) [inline bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/monoHiggs/2HDMa_UL/2HDMa_gg_sinp_theta_tanb_beta_mXd_10_MH3_YY_MH4_XX_MH2_YY_MHC_YY/2HDMa_gg_sinp_theta_tanb_beta_mXd_10_MH3_YY_MH4_XX_MH2_YY_MHC_YY_run_card.dat:31]: change it to 5000

**Review by menglu21 — COMMENTED** [review]
besides the comments inline, could you please remove the modifications on the central script (although they seems do nothing)

**Comment by shdutta16** [comment]
> besides the comments inline, could you please remove the modifications on the central script (although they seems do nothing)

I had made changes to the scripts for making the gridpacks, but after that I reverted all those changes back to what it was in the central repository. That's why the modifications do nothing.

File: genfragments/ThirteenTeV/monoHiggs_UL/pythia8_hadronizer_nomatching_Haa_cff_CP5_PSweights_2018.py (hunk 1 above)
Comment (shdutta16) [reply genfragments/ThirteenTeV/monoHiggs_UL/pythia8_hadronizer_nomatching_Haa_cff_CP5_PSweights_2018.py:6]: Can you please elaborate on this point? I do not fully understand what you meant.

File: bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/monoHiggs/2HDMa_UL/2HDMa_bb_sinp_theta_tanb_beta_mXd_10_MH3_YY_MH4_XX_MH2_YY_MHC_YY/2HDMa_bb_sinp_theta_tanb_beta_mXd_10_MH3_YY_MH4_XX_MH2_YY_MHC_YY_run_card.dat (hunk 2 above)
Comment (shdutta16) [reply bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/monoHiggs/2HDMa_UL/2HDMa_bb_sinp_theta_tanb_beta_mXd_10_MH3_YY_MH4_XX_MH2_YY_MHC_YY/2HDMa_bb_sinp_theta_tanb_beta_mXd_10_MH3_YY_MH4_XX_MH2_YY_MHC_YY_run_card.dat:31]: This is done. I have put 5000.

File: bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/monoHiggs/2HDMa_UL/2HDMa_gg_sinp_theta_tanb_beta_mXd_10_MH3_YY_MH4_XX_MH2_YY_MHC_YY/2HDMa_gg_sinp_theta_tanb_beta_mXd_10_MH3_YY_MH4_XX_MH2_YY_MHC_YY_run_card.dat (hunk 3 above)
Comment (shdutta16) [reply bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/monoHiggs/2HDMa_UL/2HDMa_gg_sinp_theta_tanb_beta_mXd_10_MH3_YY_MH4_XX_MH2_YY_MHC_YY/2HDMa_gg_sinp_theta_tanb_beta_mXd_10_MH3_YY_MH4_XX_MH2_YY_MHC_YY_run_card.dat:31]: I have changed to 5000

File: genfragments/ThirteenTeV/monoHiggs_UL/pythia8_hadronizer_nomatching_Haa_cff_CP5_PSweights_2018.py (hunk 1 above)
Comment (menglu21) [reply genfragments/ThirteenTeV/monoHiggs_UL/pythia8_hadronizer_nomatching_Haa_cff_CP5_PSweights_2018.py:6]: it's Pythia8ConcurrentHadronizerFilter

**Comment by sarafiorendi** [comment]
hi @menglu21,
a gentle ping to possibly merge this PR if no issues are found.
thanks!

**Comment by sarafiorendi** [comment]
hi @lviliani @bbilin 
could you please review/merge this PR?
thanks in advance!

**Comment by lviliani** [comment]
Hi, sorry for the delay.
Can you please clarify a couple of points?
1. Did you actually change `nevents` to 5000 as suggested by @menglu21? I still see the previous number in the cards.
2. You are adding new cards in the `monoHiggs/2HDMa_UL` directory, and also modifying something in the `monoHiggs_2HDMa_UL` directory? Is this done on purpose?

**Comment by shdutta16** [comment]
> Hi, sorry for the delay. Can you please clarify a couple of points?
> 
>     1. Did you actually change `nevents` to 5000 as suggested by @menglu21? I still see the previous number in the cards.
> 
>     2. You are adding new cards in the `monoHiggs/2HDMa_UL` directory, and also modifying something in the `monoHiggs_2HDMa_UL` directory? Is this done on purpose?

Sorry, I have made a mix-up with the MonoH(bb)-group request due to very similar names. 

1. I had made the changes for the `nevents` to 5000 in my area, but due to the mix-up haven't pushed the changes in the proper place.
2. `monoHiggs_2HDMa_UL` is the directory created by the MonoH(bb) group. By mistake, I have pushed changes from the updates in this area.  `monoHiggs/2HDMa_UL`  is the directory created by me, where I was supposed to push all the changes. Let me sort this out and then get back to you.

**Comment by shdutta16** [comment]
> Hi, sorry for the delay. Can you please clarify a couple of points?
> 
>     1. Did you actually change `nevents` to 5000 as suggested by @menglu21? I still see the previous number in the cards.
> 
>     2. You are adding new cards in the `monoHiggs/2HDMa_UL` directory, and also modifying something in the `monoHiggs_2HDMa_UL` directory? Is this done on purpose?

Sorry, I have made a mix-up with the MonoH(bb)-group request due to very similar names.

1. Yes, I had changed `nevents` to 5000 but due to the mix-up didn't push the changes in the correct area. 
2. The directory `monoHiggs_2HDMa_UL` is created by the MonoH(bb)-group. By mistake, I pushed most of my changes here in the later commits. `monoHiggs/2HDMa_UL` is the directory I had created, where I was supposed to push the change. Let me sort this out and get back to you.

**Comment by shdutta16** [comment]
Hello @lviliani, 

I have reverted all the changes made to ```monoHiggs_2HDMa_UL``` (monoH(bb)-request) and have pushed all my changes to ```monoHiggs/2HDMa_UL``` (monoH(gg)-request). Please, let me know if you see any more issues with the PR.

Thanks.

**Comment by lviliani** [comment]
Thanks! Looks good now, I'm merging it.

**Pull request merged**